### PR TITLE
Add JDBC URL sanitizer to prevent credential exposure in logs

### PR DESCRIPTION
**Pull request by gopalldb**
## Summary
Implements a comprehensive JDBC URL sanitizer to prevent accidental exposure of sensitive credentials (passwords, tokens, secrets) in log files and exception messages.

## Changes
- **Created SecurityUtil class** with `sanitizeJdbcUrl()` method that redacts all sensitive parameters
- **Updated exception messages** in Driver.java and DatabricksConnectionContext.java to sanitize URLs
- **Updated test files** (LoggingTest.java, SSLTest.java) to use sanitizer as best practice
- **Added comprehensive unit tests** - 20 tests covering all credential types and edge cases

## Sensitive Parameters Covered
The sanitizer redacts 15+ credential types from DatabricksJdbcUrlParams:
- PASSWORD, PWD
- CLIENT_SECRET, OAUTH2SECRET
- AUTH_ACCESS_TOKEN
- AUTH_REFRESH_TOKEN, OAUTH_REFRESH_TOKEN
- PROXY_PWD, CF_PROXY_PWD
- PROXY_USER, PROXY_UID, CF_PROXY_UID
- JWT_PASS_PHRASE, AUTH_JWT_KEY_PASSPHRASE
- SSL_TRUST_STORE_PASSWORD, SSL_KEY_STORE_PASSWORD
- TOKEN_CACHE_PASS_PHRASE
- UID

## Test Results
All 20 unit tests pass successfully:
```
Tests run: 20, Failures: 0, Errors: 0, Skipped: 0
```

## Example
Before:
```
jdbc:databricks://host:443/default;PWD=secret123;UID=user@example.com;HttpPath=/sql/1.0
```

After:
```
jdbc:databricks://host:443/default;PWD=***REDACTED***;UID=***REDACTED***;HttpPath=/sql/1.0
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main/java/com/databricks/jdbc/common/util/SecurityUtil.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   * <p>Pattern matches: - Parameter name (case-insensitive) - Equals sign - Value (everything until
+   * semicolon, ampersand, or end of string)
+   */
+  private static final Pattern CREDENTIAL_PATTERN =
+      Pattern.compile(
+          "(PASSWORD|PWD"
+              + "|CLIENT_?SECRET|OAUTH2SECRET"
+              + "|AUTH_?ACCESS_?TOKEN"
+              + "|AUTH_?REFRESH_?TOKEN|OAUTH_?REFRESH_?TOKEN"
+              + "|PROXY_?PWD|CF_?PROXY_?PWD"
```
Comment (vikrantpuppala) [inline src/main/java/com/databricks/jdbc/common/util/SecurityUtil.java]: can we limit to actual values in URL params, for ex: there is no proxy_pwd, only proxypwd. similar for other like truststore/keystore where the key is SSLTrustStorePwd but we check for underscores, full form of password etc.

File: src/main/java/com/databricks/jdbc/common/util/SecurityUtil.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+  /**
+   * Checks if a parameter name represents a credential or sensitive value.
+   *
+   * <p>This method can be used to determine whether a parameter should be redacted before logging.
+   *
+   * @param parameterName the parameter name to check, case-insensitive
+   * @return true if the parameter represents a credential, false otherwise
+   */
+  public static boolean isCredentialParameter(String parameterName) {
```
[the remark this answers is not in the document]
Comment (gopalldb) [reply src/main/java/com/databricks/jdbc/common/util/SecurityUtil.java]: Removed
